x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeSourceHandler.java

```diff
@@ -203,6 +203,7 @@ void onSinkFailed(Exception originEx) {
                 }
                 return first;
             });
+            buffer.waitForReading().onResponse(null); // resume the Driver if it is being blocked on reading
```

Comment:
We can notify about the failure, but I think it's simpler just to resume and let the driver handle the error, as if it hadn't been blocked before.